ast/marshal_test.go

```diff
@@ -0,0 +1,1329 @@
+				if comment.Location == nil {
+					t.Fatal("expected location to be non-nil")
+				}
+
+				// comment.Equal will check the location too
+				if !comment.Equal(data.ExpectedComment) {
+					equal = false
+				}
+			} else {
+				if !bytes.Equal(comment.Text, data.ExpectedComment.Text) {
+					equal = false
+				}
+			}
+			if !equal {
+				t.Fatalf("expected:\n%#v got\n%#v", data.ExpectedComment, comment)
+			}
+		})
+	}
+}
+
+func TestImport_MarshalJSON(t *testing.T) {
+	testCases := map[string]struct {
+		Import       *Import
+		ExpectedJSON string
+	}{
+		"base case": {
+			Import: func() *Import {
+				v, _ := InterfaceToValue("example")
+				term := Term{
+					Value:    v,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				}
+				return &Import{Path: &term}
+			}(),
+			ExpectedJSON: `{"path":{"type":"string","value":"example"}}`,
+		},
+		"location excluded": {
+			Import: func() *Import {
+				v, _ := InterfaceToValue("example")
+				term := Term{
+					Value:    v,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				}
+				return &Import{
+					Path:     &term,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+					jsonFields: map[string]bool{
+						"location": false,
+					},
+				}
+			}(),
+			ExpectedJSON: `{"path":{"type":"string","value":"example"}}`,
+		},
+		"location included": {
+			Import: func() *Import {
+				v, _ := InterfaceToValue("example")
+				term := Term{
+					Value:    v,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				}
+				return &Import{
+					Path:     &term,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+					jsonFields: map[string]bool{
+						"location": true,
+					},
+				}
+			}(),
+			ExpectedJSON: `{"location":{"file":"example.rego","row":1,"col":2},"path":{"type":"string","value":"example"}}`,
+		},
+	}
+
+	for name, data := range testCases {
+		t.Run(name, func(t *testing.T) {
+			bs := util.MustMarshalJSON(data.Import)
+			got := string(bs)
+			exp := data.ExpectedJSON
+
+			if got != exp {
+				t.Fatalf("expected:\n%s got\n%s", exp, got)
+			}
+		})
+	}
+}
+
+func TestImport_UnmarshalJSON(t *testing.T) {
+	testCases := map[string]struct {
+		JSON           string
+		ExpectedImport *Import
+	}{
+		"base case": {
+			JSON: `{"path":{"type":"string","value":"example"}}`,
+			ExpectedImport: func() *Import {
+				v, _ := InterfaceToValue("example")
+				term := Term{
+					Value: v,
+				}
+				return &Import{Path: &term}
+			}(),
+		},
+		"location case": {
+			JSON: `{"location":{"file":"example.rego","row":1,"col":2},"path":{"type":"string","value":"example"}}`,
+			ExpectedImport: func() *Import {
+				v, _ := InterfaceToValue("example")
+				term := Term{
+					Value:    v,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				}
+				return &Import{
+					Path:     &term,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+					jsonFields: map[string]bool{
```

Comment:
Nit: The other tests do not include the `jsonFields` field in the output which is fine I guess.

Reply:
Yeah, that's a good spot. It's not needed and is inconsistent. I have removed it in 8e2fe4e0510e9094823ee7279143d5971a2bb4bb